nautobot/dcim/tests/integration/test_device_bulk_edit.py

```diff
@@ -0,0 +1,181 @@
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:location_list"))
+        self.click_list_view_add_button()
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:location_add"))
+        self.fill_select2_field("location_type", "Test Location Type 1")
+        self.fill_select2_field("status", "")  # pick first status
+        self.browser.fill("name", "Test Location 2")
+        self.click_edit_form_create_button()
+
+        # Role
+        self.click_navbar_entry("Organization", "Roles")
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("extras:role_list"))
+        self.click_list_view_add_button()
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("extras:role_add"))
+        self.browser.fill("name", "Test Role 1")
+        self.fill_select2_multiselect_field("content_types", "dcim | device")
+        self.click_edit_form_create_button()
+
+    def _create_device(self, name, location="Test Location 1"):
+        self.click_navbar_entry("Devices", "Devices")
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:device_list"))
+        self.click_list_view_add_button()
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:device_add"))
+        self.browser.fill("name", name)
+        self.fill_select2_field("role", "Test Role 1")
+        self.fill_select2_field("device_type", "Test Device Type 1")
+        self.fill_select2_field("location", location)
+        self.fill_select2_field("status", "")  # pick first status
+        self.click_edit_form_create_button()
+
+    def tearDown(self):
+        self.logout()
+        super().tearDown()
+
+    def _go_to_devices_list(self):
+        self.click_navbar_entry("Devices", "Devices")
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:device_list"))
+
+    def test_bulk_edit_require_selection(self):
+        self._go_to_devices_list()
+
+        # Click "edit selected" without selecting anything
+        self.click_bulk_edit()
+
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:device_list"))
+        self.assertTrue(self.browser.is_text_present("No devices were selected.", wait_time=5))
+
+    def test_bulk_edit_all_devices(self):
+        # Create device for test
+        self._create_device("Test Device Integration Test 1")
+        self._create_device("Test Device Integration Test 2")
+        self._go_to_devices_list()
+
+        # Select all devices and edit them
+        self.select_all_items()
+        self.click_bulk_edit()
+        self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:device_bulk_edit"))
+
+        # Submit bulk edit form without any changes
+        self.submit_bulk_edit_operation()
+
+        # Verify job output
+        self.assertIsBulkEditJob()
+        self.assertJobStatusIsCompleted()
```

Comment:
Can we use the ORM to check which devices were or were not updated in each of these tests?